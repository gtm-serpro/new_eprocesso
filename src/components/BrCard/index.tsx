import { CSSProperties } from "react";
import classNames from "classnames";
import BrButton from "../BrButton";
import Icon from "@/common/Icon";
import BrAvatar, { BrAvatarProps } from "../BrAvatar";

export type BrCardProps = {
  children: React.ReactNode;
  title?: string;
  subtitle?: string;
  avatar?: BrAvatarProps;
  footer?: React.ReactNode;
  disabled?: boolean;
  hover?: boolean;
  fixedHeight?: boolean;
  style?: CSSProperties;
  className?: string;
  onClick?: () => void;
};

const BrCard = (props: BrCardProps) => {
  const {
    children,
    className,
    style,
    title,
    subtitle,
    avatar,
    footer,
    disabled,
    hover,
    fixedHeight,
    onClick,
  } = props;

  const classesRoot = classNames("br-card", { disabled, hover, "h-fixed": fixedHeight }, className);

  const classesTitle = classNames({ "ml-3": !!avatar });

  const showHeader = !!title || !!subtitle || !!avatar;

  return (
    <div className={classesRoot} onClick={onClick} style={style}>
      {showHeader && (
        <div className="card-header">
          <div className="d-flex">
            {avatar && <BrAvatar {...avatar} />}
            <div className={classesTitle}>
              {title && <div className="text-weight-semi-bold text-up-02">{title}</div>}
              {subtitle && <div>{subtitle}</div>}
            </div>
            <div className="ml-auto">
              <BrButton circle>
                <Icon icon="ellipsis-v" aria-hidden="true" />
              </BrButton>
            </div>
          </div>
        </div>
      )}
      <div className="card-content">{children}</div>
      {footer && <div className="card-footer">{footer}</div>}
    </div>
  );
};

export default BrCard;
