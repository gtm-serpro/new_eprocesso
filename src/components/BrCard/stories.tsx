import BrButton from "@/components/BrButton";
import { Meta, Story } from "@storybook/react";
import BrCard, { BrCardProps } from ".";

export default {
  title: "Superfícies/Card",
  component: BrCard,
} as Meta;

export const Simples: Story<BrCardProps> = (args) => <BrCard {...args}>Conteúdo</BrCard>;

export const Completo: Story<BrCardProps> = (args) => (
  <BrCard
    avatar={{ title: "título do avatar" }}
    footer={
      <BrButton primary size="small">
        Conteúdo Rodapé
      </BrButton>
    }
    {...args}
  >
    Conteúdo
  </BrCard>
);
Completo.args = { title: "Título", subtitle: "Subtítulo" };
