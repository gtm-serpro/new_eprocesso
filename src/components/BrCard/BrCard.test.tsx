import { render, screen } from "@testing-library/react";
import BrCard from ".";

describe("BrCard", () => {
  it("deveria renderizar conteúdo do BrCard", () => {
    const { container } = render(<BrCard>Conteúdo</BrCard>);

    expect(screen.getByText("Conteúdo")).toHaveClass("card-content");

    expect(container.firstChild).toMatchSnapshot();
  });

  it("deveria renderizar título do BrCard", () => {
    render(<BrCard title="Título">Conteúdo</BrCard>);
    expect(screen.getByText("Título")).toHaveClass("text-weight-semi-bold text-up-02");
  });

  it("deveria renderizar root do BrCard", () => {
    const { container } = render(<BrCard>Conteúdo</BrCard>);
    expect(container.firstChild).toHaveClass("br-card");
  });

  it("deveria renderizar desabilitado", () => {
    const { container } = render(<BrCard disabled>Conteúdo</BrCard>);
    expect(container.firstChild).toHaveClass("br-card");
    expect(container.firstChild).toHaveClass("disabled");
  });

  it("deveria renderizar o footer", () => {
    render(<BrCard footer={<div className="test-footer">Footer</div>}>Conteúdo</BrCard>);
    const footer = screen.getByText("Footer");
    expect(footer).toHaveClass("test-footer");
    expect(footer.parentElement).toHaveClass("card-footer");
  });
});
