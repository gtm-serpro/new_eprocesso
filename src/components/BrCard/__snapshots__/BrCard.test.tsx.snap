// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrCard deveria renderizar conteúdo do BrCard 1`] = `
<div
  class="br-card"
>
  <div
    class="card-content"
  >
    Conteúdo
  </div>
</div>
`;
